072-DataScienceInFabric/Student/Resources/deployhack.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+read -p "Enter a prefix for your resource group and resources: " RESOURCEPREFIX
+response=$(az storage account check-name --name $RESOURCEPREFIX)
+available=$(jq -n --argjson data "$response" '$data.nameAvailable')
+
+until [[ $available == "true" ]]
+do
+    echo "'$RESOURCEPREFIX' is not available as a storage account name. Please try a different name"
+    read -p "Enter a prefix for your resource group and resources: " RESOURCEPREFIX
+    response=$(az storage account check-name --name $RESOURCEPREFIX)
+    available=$(jq -n --argjson data "$response" '$data.nameAvailable')
+    
+done
+
+RGNAME="$RESOURCEPREFIX-rg"
+deploymentName="$RESOURCEPREFIX-deployment"
+
+read -p "Enter an Azure region to deploy to (i.e. 'eastus', 'westus', 'northeurope'): " LOCATION
+
+echo "Creating resource group '$RGNAME' in Azure region '$LOCATION'..."
+az group create --location $LOCATION --name $RGNAME
+
+echo "Deploying Azure Storage Account where your training data will be located..."
+az storage account create --name $RESOURCEPREFIX --resource-group $RGNAME --sku Standard_LRS
+
+echo "Validating storage account for ADLS Gen 2 migration... "
+az storage account hns-migration start --request-type validation --name $RESOURCEPREFIX
+
+echo "Upgrading the Azure Storage Account to ADLS gen 2 to enable the creation of shortcuts..."
+az storage account hns-migration start --request-type upgrade --name $RESOURCEPREFIX
+
+echo "Downloading the required data from the GitHub repo..."
+wget https://raw.githubusercontent.com/pradeepsingla87/WhatTheHack/xxx-DataScience_In_MicrosoftFabric/072-DataScienceInFabric/Student/Resources/heartdata.csv
```

Comment:
@pradeepsingla87 @juanlldc @lesantana, should the URL for the location of the CSV reflect the WTH repository?

Reply:
Spoke with Juan and this will be updated post publish